ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/backup/view/BackupView.kt

```diff
@@ -72,32 +89,78 @@ fun BackupWallet(
-        BackupStage.EducationRecoveryPhrase -> EducationRecoveryPhrase(onNext = backupState::goNext)
-        BackupStage.Seed -> SeedPhrase(
-            wallet,
-            onNext = backupState::goNext,
-            onCopyToClipboard = onCopyToClipboard
-        )
-        BackupStage.Test -> Test(
-            wallet,
-            choices,
-            onBack = backupState::goPrevious,
-            onNext = backupState::goNext,
+    val currentBackupStage = backupState.current.collectAsState().value
+
+    Scaffold(
+        topBar = {
+            BackupTopAppBar(
+                backupStage = currentBackupStage,
+                onCopyToClipboard = onCopyToClipboard,
+                onBack = backupState::goPrevious,
+                selectedTestChoices = choices
+            )
+        },
+        bottomBar = {
+            BackupBottomNav(
+                backupStage = currentBackupStage,
+                onNext = backupState::goNext,
+                onBack = backupState::goPrevious,
+                selectedTestChoices = choices,
+                onComplete = onComplete,
+                onBackToSeedPhrase = {
+                    backupState.goToStage(BackupStage.CheckSeed)
+                }
+            )
+        }
+    ) { paddingValues ->
+        BackupMainContent(
+            paddingValues = paddingValues,
+            backupState = backupState,
+            wallet = wallet,
+            choices = choices,
             onChoicesChanged = onChoicesChanged
         )
-        BackupStage.Complete -> Complete(
-            onComplete = onComplete,
-            onBackToSeedPhrase = backupState::goToSeed
-        )
     }
 }
 
 @Composable
-private fun EducationOverview(onNext: () -> Unit) {
-    Column(modifier = Modifier.verticalScroll(state = rememberScrollState())) {
-        Header(stringResource(R.string.new_wallet_1_header))
+fun BackupMainContent(
+    paddingValues: PaddingValues,
+    backupState: BackupState,
+    wallet: PersistableWallet,
+    choices: TestChoices,
+    onChoicesChanged: ((choicesCount: Int) -> Unit)?
+) {
+    Column(
+        Modifier
+            .padding(
+                top = paddingValues.calculateTopPadding(),
+                bottom = paddingValues.calculateBottomPadding()
+            )
+    ) {
+        when (backupState.current.collectAsState().value) {
+            is BackupStage.EducationOverview -> EducationOverview()
+            is BackupStage.EducationRecoveryPhrase -> EducationRecoveryPhrase()
+            is BackupStage.Seed -> SeedPhrase(wallet)
+            is BackupStage.Test -> TestInProgress(
+                selectedTestChoices = choices,
+                onChoicesChanged = onChoicesChanged,
+                splitSeedPhrase = wallet.seedPhrase.split,
+                backupState = backupState
+            )
+            is BackupStage.Failure -> TestFailure()
+            is BackupStage.Complete -> TestComplete()
+            is BackupStage.CheckSeed -> SeedPhrase(wallet)
```

Comment:
Fine for now, but note that in the future we'll probably copy-paste the SeedPhrase composable to have one complex version showing Google Credential manager choices and a simple version for re-review at the end.